sec_certs/bsi_parse.py

```diff
@@ -136,23 +101,23 @@ class Bsitmp(ComplexSerializableType):
     valid_until: str
     soup: BeautifulSoup
     id: str
+    pdf_links: list[str]
 
     def __init__(self, url):
         # self.html_content = pd.read_html(url)
         # self.certification_date = self.html_content[0].to_numpy()[3][1]
         # self.valid_until = self.html_content[0].to_numpy()[4][1]
         self.soup = BeautifulSoup(requests.get(url).content, "html.parser")
         self.id = self.soup.find("meta", property='title')
+        self.pdf_links = \
+            ["https://www.bsi.bund.de/" + a['href']
+             for a in self.soup.find_all('a', href=True, recursive=True, class_='RichTextDownloadLink Publication FTpdf')]
+        # for a in self.soup.find_all('a', href=True, recursive=True, class_='RichTextDownloadLink Publication FTpdf'):
+        #     self.pdf_links.append("https://www.bsi.bund.de/" + a['href'])
 
 
-subpage = Bsitmp(test_url)
-
-
-# print(subpage.certification_date)
-
-
-def process(root_url: str):
-    browser = BsiBrowser(root_url)
+def process(base_url: str):
+    browser = BsiBrowser(base_url)
     browser.parse()
     tmp_list = []
     for handler in browser.handler_list:
```

Comment:
This whole for loop can be written as a list comprehension and you can call `tmp_list.extend([the_comprehension])`